frontend/src/services/ErrorHandler.ts

```diff
@@ -0,0 +1,36 @@
+import {Dispatch} from 'redux'
+import { toast } from 'react-toastify'
+
+import {errorMessage } from '../constants/errorAndSuccessMessages'
+import {userPreferencesRequestFailed} from '../actions/prefAndProfileActions'
+
+export interface IError {
+        field: string
+        message: string
+        statusCode?: number // код статуса ошибки например 401 или 500
+        errorType?: string // тип ошибки например ошибка формы
+}
+
+export interface IErrorHandlerReturn {
+    errorText: string | string[]
+}
+
+export const errorHandler = (error: IError, dispatch: Dispatch): void => {
```

Comment:
I would recommend to get rid of redux here, and instead of `dispatch` provide a function that at least consume error message.